oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -691,23 +691,40 @@ int mca_spml_ucx_add_procs(oshmem_group_t* group, size_t nprocs)
         }
     }
 
+    indices = malloc(nprocs * sizeof(int));
+    if (!indices) {
+        goto error;
+    }
+
+    for (i = 0; i < nprocs; i++) {
+        indices[i] = i;
```

Comment:
IMO can omit this init and use `i` in line 710 instead.

Reply:
if i get the algo right, i think we cannot because we need to track the previously shuffled indices.